src/tribler/core/components/torrent_checker/torrent_checker/torrent_checker.py

```diff
@@ -197,27 +196,35 @@ def load_torrents_checked_from_db(self):
     @db_session
     def torrents_to_check(self):
         """
-        Two categories of torrents are selected (popular & old). From the pool of selected torrents, a certain
-        number of them are submitted for health check. The torrents that are within the freshness window are
-        excluded from the selection considering the health information is still fresh.
-
-        1. Popular torrents (50%)
-        The indicator for popularity here is considered as the seeder count with direct proportionality
-        assuming more seeders -> more popular. There could be other indicators to be introduced later.
-
-        2. Old torrents (50%)
-        By old torrents, we refer to those checked quite farther in the past, sorted by the last_check value.
+        A single popular and a single old torrent is selected.
+        Selection randomness is based on normal variate distribution so that for popular torrent selection,
+        the probability of selecting a popular torrent is higher and likewise for old torrent selection.
         """
         last_fresh_time = time.time() - HEALTH_FRESHNESS_SECONDS
-        popular_torrents = list(self.mds.TorrentState.select(lambda g: g.last_check < last_fresh_time).
-                                order_by(lambda g: (desc(g.seeders), g.last_check)).limit(TORRENT_SELECTION_POOL_SIZE))
+        total_torrents = self.mds.TorrentState.select().count()
+        random_index = self._select_randomly_likely_popular_torrent_index(total_torrents)
+
+        random_popular_torrent = self.mds.TorrentState.select(lambda g: g.last_check < last_fresh_time)\
+                                                      .order_by(lambda g: (desc(g.seeders), g.last_check))\
+                                                      .fetch(limit=1, offset=random_index)[:]
 
-        old_torrents = list(self.mds.TorrentState.select(lambda g: g.last_check < last_fresh_time).
-                            order_by(lambda g: (g.last_check, desc(g.seeders))).limit(TORRENT_SELECTION_POOL_SIZE))
+        random_old_torrent = self.mds.TorrentState.select(lambda g: g.last_check < last_fresh_time)\
+                                                  .order_by(lambda g: (g.last_check, desc(g.seeders)))\
+                                                  .fetch(limit=1, offset=random_index)[:]
```

Comment:
I think it is not a good idea to use normally distributed offsets when selecting random torrents from the database for the check.

Let's consider a real-life situation when a user subscribed to several popular channels and has 3.5 million torrents in the database. In this case, the average offset will be approximately 1 million. So, to select a random torrent from the database, the SQL engine should sort torrents by `last_check` and `seeders` columns and then skip the first one million torrents to select a random one for the check. On a decent machine with an SSD drive, it took around 2 seconds just to select a single random torrent. During that time, the event loop is blocked, and it may be much slower on a machine with a traditional HDD drive.

It is not as necessary to select random torrents to check in a perfectly normalized way. Instead, we can select N random torrents from the database and choose the most popular from these torrents. It should be much faster. On the same machine, the query `self.mds.TorrentState.select_random(100)` takes four milliseconds for execution, much better than 2 seconds for the original query.

So, I suggest doing the following instead:
1. Select N random torrents
2. From these torrents, in Python, filter torrents with `torrent.last_check < last_fresh_time` (most torrents from the database should satisfy this criteria)
3. Order the resulting list by the number of seeders, and choose the most popular one.

Something like that (not tested):
```python
random_torrents = self.mds.TorrentState.select_random(100)
old_torrents = [t for t in random_torrents if t.last_check < last_fresh_time]
result = []
if old_torrents:
    random_old_torrent = old_torrents[0]
    result.append(random_old_torrent)
    if len(old_torrents) > 1:
        popular_random_torrents = sorted(old_torrents[1:], key = lambda torrent: torrrent.seeders, reverse=True)
        most_popular_random_torrent = popular_random_torrents[0]
        result.append(most_popular_random_torrent)
return result    
```

Reply:
Thank you for the opinion and the suggestion. I think this needs more in-depth thinking and thus can be its own separate PR that only deals with the changes to the torrent checker and not with the Popularity community. Therefore, I'll remove these changes from PR.